safe-modules/liveness-module-v2.md

```diff
@@ -0,0 +1,139 @@
+
+The customers for this design doc are any participants in the multisig accounts that would utilize
+this module, as well as any 3rd parties who rely on the security properties of these accounts.
+
+### Customer Reviewers
+
+<!-- Identify at least one customer who should be involved in the review of this document. -->
+
+- Coinbase (potentially impacted 3rd party)
+- Uniswap (`LivenessModule` user)
+- Optimism Foundation (`LivenessModule` user)
+
+## Requirements and Constraints
+
+<!-- Identify the solution requirements and any additional design constraints from the Context and
+Problem Statement section in a bulleted list. -->
+
+- Solve both the owner liveness and account liveness problems *securely*.
+- Keep code as minimal as possible to avoid complex logic in Safe modules.
+- Reduce mental overhead for Safe owners and multisig leads.
+
+## Proposed Solution
+
+<!-- Explain the solution that you believe best addresses the problem described above. -->
+
+We propose the introduction of a new module, `LivenessModuleV2` (name is unimportant) to replace
+the existing `LivenessGuard`/`LivenessModule`. `LivenessModuleV2` would employ a challenge/response
+mechanism as an alternative to the existing poke mechanism inside of `LivenessGuard`.
+
+### Singleton
+
+`LivenessModuleV2` is a singleton that exists at a known address on all chains and has no
+constructor parameters. Any Safe on any network can choose to configure the module and then enable
+the module within the Safe.
+
+### Configuration
+
+A Safe configures the module by setting a challenge period, a challenge bond, and a fallback owner.
+
+### Challenge Protocol
+
+Any user may trigger an Account Liveness Challenge by targeting a specific Safe address. In
+response to this challenge, a threshold of owners (same threshold as the Safe itself) must sign a
+message that confirms the liveness of the safe (e.g., a message over the Safe address and the
+current timestamp). If the Safe fails to reach this threshold, all owners are removed from the
+account and ownership is transferred to the fallback account. If the Safe successfully reaches the
+threshold, the challenge fails and the Safe receives the bond as a reward.
+
+The Account Liveness Challenge addresses both of the ways in which a Safe can fail to be live. If
+sufficiently many honest users cannot sign that the account cannot reach a threshold, the account
+falls back to the fallback owner. If sufficiently many users are malicious that the account cannot
+reach a threshold, the honest users can refuse to sign and force a fallback.
+
+## Alternatives Considered
+
+<!-- Describe any alternatives that were considered during the development of this design. Explain
+why the alternative designs were ultimately not chosen and where they failed to meet the product
+requirements. -->
+
+### Owner Liveness Challenge
+
+We previously considered a model that would allow users to challenge either the entire Safe (as
```

Comment:
We don't describe the alternative model where the entire Safe is challenged, and it is confusing because in this design doc we are proposing a module where the entire Safe is challenged.

Was the Account Liveness Challenge like the Owner Liveness Challenge, but in a loop so that it challenges all owners in turn?

In any case, I would just mention the Owner Liveness Challenge here. I think that's enough.